### ISSUE TITLE
Concurrent container auto remove orphans containers on vSphere

### ISSUE THREAD

**Issue by cgtexmex** [vmware/vic#6342]
I created 16 containers concurrently with --rm specified and named 2 - 17 via:

`docker run -d --rm --name {#} ubuntu /bin/sh -c 'a=0; while [ $a -lt 75 ]; do echo "line $a"; a=`expr $a + 1`; sleep 2; done;'`

This resulted in 16 containers:
```
docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED              STATUS              PORTS               NAMES
b07f1ab1d37a        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       5
bb7c09c14676        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       4
111fb3a28833        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       12
2c733c90a322        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       13
5062fadd9df3        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       15
15ba4f59d9ba        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       11
fee7c5d822b8        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       2
c03a83fb6a85        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       7
f81b635bceca        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       3
4c672443ec4d        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       6
6a1cb2bb5636        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       16
b0e0de57d268        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       10
6d929e8800ef        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       17
bf9edaf8f983        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       14
29ff1eb418ea        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       9
ebca47a44ccf        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       8
```
The command provided to the container will run for roughly 2.5 minutes, so the containers will all exit at the same time.

All but 2 containers were removed (as detailed in #6341):
```
docker ps -a
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS                      PORTS               NAMES
b07f1ab1d37a        ubuntu              "/bin/sh -c a=0; w..."   20 minutes ago      Stopped                                         5
2c733c90a322        ubuntu              "/bin/sh -c a=0; w..."   20 minutes ago      Exited (0) 16 minutes ago                       13
```

Additionally, one container was orphaned on vSphere.  Meaning it was removed from the docker persona, but the containerVM wasn't removed:
```
govc ls vm
/vcqaDC/vm/Discovered virtual machine
/vcqaDC/vm/9-29ff1eb418ea
/vcqaDC/vm/victor
```
Environment details:
vSphere 6.0 u3 (nimbus)
ESXi 6.0 u3
Storage: nfs
vic 1.2.1 deployed to resource pool via `--use-rp`
Reproducibility: easy

[logs](https://github.com/vmware/vic/files/1304701/orphaned_network.zip)

**Comment by hickeng** [comment]
@cgtexmex we've got some vc-support bundles pending - we've seen entries being resurrected in the VC inventory but that don't have a vmx or other VM files on the datastore. We've also seen discovery of invisible inventory entries when running `vic-machine delete` for cVMs that don't show up in the UI. We only noticed this because there's a warning that they failed the destroy and are being unregistered as an alternative.
Can you confirm next time you see this whether those orphaned VMs are actually fully functional with files on the datastore or are only inventory entries.

**Comment by cgtexmex** [comment]
@hickeng I've updated the initial comment to show that it's NFS storage.  I was able to recreate it today and the VM is indeed real.  I restarted the VCH so it would find the containerVM and then started it via `docker start`.

So a workaround is to restart the VCH, but a developer most likely won't have VC / govc access and wouldn't know that it's orphaned.

**Comment by hickeng** [comment]
Added #6341 as a dependency of this so that we start with investigating a specific portion of this behaviour. #6341 seems the most tractable at a glance so we'll start with that - re-assess as needed once investigation is underway.

**Comment by cgtexmex** [comment]
Ensure we see no appearance of #6266 during resolution of this issue

**Comment by cgtexmex** [comment]
Executed the following in parallel for 20 Container VMS:
```
docker create --name {number} --rm ubuntu /bin/sh -c 'a=0; while [ $a -lt 150 ]; do echo "line $a"; a=`expr $a + 1`; sleep 2; done;'
docker start {number}
```
This will create a VM that loops for 5 minutes and then powerOff.  The powerOff should trigger the `--rm` based on container event activity.

Result:
All 20 containers were removed from vic, but 1 container VM remained on vSphere.  Only users with vSphere access would be able to see the remaining container.  

Cause:
The vm in question was migrated during it's lifecycle.  Once the vm was migrated to it's new host the originating host executed a clean up routine to remove the the `.vmx` file -- that resulted in a vSphere VM Remove event.  That event removed the container VM from vic -- at this point vic no longer knows about the VM, but the VM is still in the cluster (albeit on a different host).

In order to remove the stranded VM one of two things must occur:

- A user with vSphere access can remove OOB
- Recycle the VCH which will allow the VCH to rediscover the _orphaned_ container..the container could now be removed via a `docker rm`


Environment Info:
Nimbus Cluster
VC 6.5 u1 (Build 5973321)
ESXi 6.5 u1 (Build 5969303) x 4

**Comment by jzt** [comment]
Seen in a blinky longevity run while attempting to create a container:

### portlayer:
It appears the reconfigure on the 3rd line takes too long resulting in a CDE while in `WaitForSession`:
```
Dec 14 2017 03:01:46.520Z INFO  op=293.253: Reconfigure: attempting update to 4473f943087b576148f78abf5d82b12590e78ede21cf3e6bf7e341e4cd193a55 with change version "2017-12-14T03:01:46.426531Z" (poweredOff)
Dec 14 2017 03:01:46.799Z INFO  op=293.253: Reconfigure: committed update to 4473f943087b576148f78abf5d82b12590e78ede21cf3e6bf7e341e4cd193a55 with change version: 2017-12-14T03:01:46.426531Z
Dec 14 2017 03:01:46.799Z INFO  op=293.253: Attempting to perform a guest reconfigure operation on (4473f943087b576148f78abf5d82b12590e78ede21cf3e6bf7e341e4cd193a55)
Dec 14 2017 03:02:04.315Z INFO  op=293.255: Reconfigure: attempting update to 4473f943087b576148f78abf5d82b12590e78ede21cf3e6bf7e341e4cd193a55 with change version "2017-12-14T03:01:50.900124Z" (poweredOn)
Dec 14 2017 03:02:04.671Z INFO  op=293.255: Reconfigure: committed update to 4473f943087b576148f78abf5d82b12590e78ede21cf3e6bf7e341e4cd193a55 with change version: 2017-12-14T03:01:50.900124Z
Dec 14 2017 03:02:04.671Z INFO  op=293.255: Attempting to perform a guest reconfigure operation on (4473f943087b576148f78abf5d82b12590e78ede21cf3e6bf7e341e4cd193a55)
Dec 14 2017 03:04:47.354Z ERROR op=293.253: vic/pkg/trace.(*Operation).Err: WaitForSession error: context deadline exceeded
vic/lib/portlayer/exec.(*Container).start:423 WaitForSession
vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).CommitHandler:218 commit handle(8c1cc458bef1058995ff2ac96b2876ce)
Dec 14 2017 03:04:47.356Z WARN  op=293.253: container and handle ChangeVersions do not match for 4473f943087b576148f78abf5d82b12590e78ede21cf3e6bf7e341e4cd193a55: 2017-12-14T03:01:50.900124Z != 2017-12-14T03:01:46.426531Z
Dec 14 2017 03:04:47.356Z ERROR op=293.253: CommitHandler error on handle(8c1cc458bef1058995ff2ac96b2876ce) for 4473f943087b576148f78abf5d82b12590e78ede21cf3e6bf7e341e4cd193a55: unable to wait for process launch status: Post https://blinky.eng.vmware.com/sdk: context deadline exceeded
```

### vpxd.log:

#### Reconfigure attempted at 3:01:46 (opID 293.253):
```
017-12-14T03:01:46.803Z info vpxd[7FB97552A700] [Originator@6876 sub=vpxLro opID=293.253-e1] [VpxLRO] -- BEGIN task-1711116 -- vm-140176 -- vim.VirtualMachine.reconfigure -- 5216048c-31cc-5fa1-bba1-bdc10aea6b48(52bbba64-e529-bf0b-d34f-c2aace313e61)
2017-12-14T03:01:47.072Z info vpxd[7FB97552A700] [Originator@6876 sub=pbm opID=293.253-e1] PBMCallback: ReconfigVmAssociations: No association to reconfig for vm vm-140176
2017-12-14T03:01:47.074Z info vpxd[7FB97552A700] [Originator@6876 sub=vpxLro opID=293.253-e1] [VpxLRO] -- FINISH task-1711116
```
#### Attempt powerOn (opID 293.253):
```
2017-12-14T03:01:47.081Z info vpxd[7FB95DEE3700] [Originator@6876 sub=vpxLro opID=293.253-7f] [VpxLRO] -- BEGIN task-1711117 -- vm-140176 -- vim.VirtualMachine.powerOn -- 5216048c-31cc-5fa1-bba1-bdc10aea6b48(52bbba64-e529-bf0b-d34f-c2aace313e61)
2017-12-14T03:01:47.086Z info vpxd[7FB95DEE3700] [Originator@6876 sub=vpxLro opID=293.253-7f-01] [VpxLRO] -- BEGIN lro-20283534 --  -- VmprovWorkflow --
```
[snip]
```
2017-12-14T03:01:47.626Z info vpxd[7FB95DEE3700] [Originator@6876 sub=vpxLro opID=293.253-7f-01] [VpxLRO] -- FINISH lro-20283534
2017-12-14T03:01:47.626Z info vpxd[7FB95DEE3700] [Originator@6876 sub=vpxLro opID=293.253-7f] [VpxLRO] -- FINISH task-1711117
```
#### powerOn task finished. I believe we ignore the event generated by this powerOn (@cgtexmex can you verify?)

#### `AskRefreshDrmRecLro` called and apprently determines there is a cluster imbalance:
```
2017-12-14T03:01:50.605Z info vpxd[7FB955DBB700] [Originator@6876 sub=vpxLro opID=lro-1-77a4044d-8633] [VpxLRO] -- BEGIN lro-20283561 --  -- AskRefreshDrmRecLro --
2017-12-14T03:01:50.611Z info vpxd[7FB955DBB700] [Originator@6876 sub=drmLogger opID=lro-1-77a4044d-8633] Drmdump written: /var/log/vmware/vpxd/drmdump/domain-c12/5564410976-proposeNonResActions.dump
2017-12-14T03:01:50.611Z warning vpxd[7FB955DBB700] [Originator@6876 sub=drmLogger opID=lro-1-77a4044d-8633] [GetNewPlacement] At end, cluster is imbalanced
2017-12-14T03:01:50.669Z info vpxd[7FB95058C700] [Originator@6876 sub=vpxLro opID=lro-1-77a4044d-8633-01] [VpxLRO] -- BEGIN lro-20283562 --  -- DrmExecute --
2017-12-14T03:01:50.670Z info vpxd[7FB94D1A5700] [Originator@6876 sub=vpxLro opID=lro-1-77a4044d-8633-02] [VpxLRO] -- BEGIN lro-20283564 --  -- DrmExecute --
2017-12-14T03:01:50.673Z info vpxd[7FB95058C700] [Originator@6876 sub=vpxLro opID=lro-1-77a4044d-8633-01] [VpxLRO] -- FINISH lro-20283562
```

#### vMotion begins at 3:01:50 (`opID=lro-1-77a4044d-8633-01-01` appears to be an extension of `opID=lro-1-77a4044d-8633-01`):
```
2017-12-14T03:01:50.676Z info vpxd[7FB95058C700] [Originator@6876 sub=vpxLro opID=lro-1-77a4044d-8633-01-01] [VpxLRO] -- BEGIN task-1711118 -- practical_mestorf-4473f943087b -- Drm.ExecuteVMotionLRO --
2017-12-14T03:01:50.677Z info vpxd[7FB95058C700] [Originator@6876 sub=drmLogger opID=lro-1-77a4044d-8633-01-01] Executing DRM recommended vMotion, migrating [vim.VirtualMachine:vm-140176,practical_mestorf-4473f943087b] from vim.HostSystem:host-62 to vim.HostSystem:host-81
2017-12-14T03:01:50.677Z info vpxd[7FB94D1A5700] [Originator@6876 sub=vpxLro opID=lro-1-77a4044d-8633-02] [VpxLRO] -- FINISH lro-20283564
2017-12-14T03:01:50.677Z info vpxd[7FB95058C700] [Originator@6876 sub=vpxLro opID=lro-1-77a4044d-8633-01-01-01] [VpxLRO] -- BEGIN lro-20283565 --  -- VmprovWorkflow --
```
[snip] - Note the vm being vMotioned is the one we attempted to create in the portlayer log: `practical_mestorf-4473f943087b`. It's being moved from krusty to quimby:
```
2017-12-14T03:01:50.678Z info vpxd[7FB95058C700] [Originator@6876 sub=VmProv opID=lro-1-77a4044d-8633-01-01-01] LocalVC Migrate of practical_mestorf-4473f943087b on vim.HostSystem:host-62(10.17.109.47) with ds ds:///vmfs/volumes/vsan:52b432a675d2bf59-7168e072a499aad9/ to vim.HostSystem:host-81(10.17.109.211) with ds ds:///vmfs/volumes/vsan:52b432a675d2bf59-7168e072a499aad9/ started
```
[snip]
```
2017-12-14T03:02:04.141Z info vpxd[7FB95058C700] [Originator@6876 sub=vpxLro opID=lro-1-77a4044d-8633-01-01-01] [VpxLRO] -- FINISH lro-20283565
```
#### vMotion finished ~15 seconds later?

### flanders vpxa.log
#### vMotion visible at 3:01:50 (10.17.109.47 resolved to flanders, 10.17.109.211 to quimby):
```
2017-12-14T03:01:50.723Z verbose vpxa[8584B70] [Originator@6876 sub=vpxaMoVMotion opID=lro-1-77a4044d-8633-01-01-01-91] [PrepareSourceEx] (7716914965410412045) Start: srcIp 10.17.109.47, dstIp 10.17.109.211, VM vm-8255, srcLoggingIp , dstLoggingIp
2017-12-14T03:01:50.723Z info vpxa[8584B70] [Originator@6876 sub=vpxaMoVMotion opID=lro-1-77a4044d-8633-01-01-01-91] [PrepareSourceEx] (7716914965410412045) Invoking IsFtFailingOverVm on VM [/vmfs/volumes/vsan:52b432a675d2bf59-7168e072a499aad9/98e9315a-e2cf-0d9c-41b4-000af7992758/practical_mestorf-4473f943087b.vmx]
```

### flanders' hostd:
```
2017-12-14T03:01:51.698Z info hostd[35781B70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/vsan:52b432a675d2bf59-7168e072a499aad9/98e9315a-e2cf-0d9c-41b4-000af7992758/practical_mestorf-4473f943087b.vmx opID=lro-1-77a4044d-8633-01-01-01-5b-e10a user=vpxuser] VMotionInitiateSrc (7716914965410412045): wid=44173783
```
```
2017-12-14T03:02:03.714Z info hostd[5AF82B70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/vsan:52b432a675d2bf59-7168e072a499aad9/98e9315a-e2cf-0d9c-41b4-000af7992758/practical_mestorf-4473f943087b.vmx] State Transition (VM_STATE_OFF -> VM_STATE_UNREGISTERING)
```
```
2017-12-14T03:02:03.715Z info hostd[5AF82B70] [Originator@6876 sub=Vimsvc.ha-eventmgr] Event 763020 : Removed practical_mestorf-4473f943087b on flanders.eng.vmware.com from ha-datacenter
```

@cgtexmex at this point the remove event is consumed and the container is removed from vic?

### quimby vpxa.log:
```
2017-12-14T03:01:50.824Z info vpxa[7EF1B70] [Originator@6876 sub=vpxLro opID=lro-1-77a4044d-8633-01-01-01-27] [VpxLRO] -- BEGIN lro-1316683 -- vmotionManager -- vim.host.VMotionManager.initiateDestination -- 52c6ea4c-5b8e-40ac-9d80-715541ae039d
2017-12-14T03:01:50.824Z verbose vpxa[7EF1B70] [Originator@6876 sub=vpxaMoVMotion opID=lro-1-77a4044d-8633-01-01-01-27] [InitiateDestination] (7716914965410412045) Start
```
[snip]
```
2017-12-14T03:01:51.167Z verbose vpxa[7F12B70] [Originator@6876 sub=halservices opID=WFU-588c74c] [VpxaHalServices] VmAdded Event for vm(0) ds:///vmfs/volumes/vsan:52b432a675d2bf59-7168e072a499aad9/98e9315a-e2cf-0d9c-41b4-000af7992758/practical_mestorf-4473f943087b.vmx
2017-12-14T03:01:51.168Z verbose vpxa[7F12B70] [Originator@6876 sub=vpxaMoVm opID=WFU-588c74c] [VpxaMoVm::CheckMoVm] did not find a VM with ID 23451 in the vmList
2017-12-14T03:01:51.168Z verbose vpxa[7F12B70] [Originator@6876 sub=vpxaMoVMotion opID=WFU-588c74c] [VpxaMoVMotion::SetVmid] Immigrating VM at path /vmfs/volumes/vsan:52b432a675d2bf59-7168e072a499aad9/98e9315a-e2cf-0d9c-41b4-000af7992758/practical_mestorf-4473f943087b.vmx has vmid 23451
2017-12-14T03:01:51.168Z verbose vpxa[7F12B70] [Originator@6876 sub=vpxaMoVm opID=WFU-588c74c] [VpxaMoVm] Created VM: practical_mestorf-4473f943087b (/vmfs/volumes/vsan:52b432a675d2bf59-7168e072a499aad9/98e9315a-e2cf-0d9c-41b4-000af7992758/practical_mestorf-4473f943087b.vmx). State: 0, FTState: 0
2017-12-14T03:01:51.168Z verbose vpxa[7F12B70] [Originator@6876 sub=vpxaInvtVm opID=WFU-588c74c] [VpxaInvtVm] Discovered new VM practical_mestorf-4473f943087b(ds:///vmfs/volumes/vsan:52b432a675d2bf59-7168e072a499aad9/98e9315a-e2cf-0d9c-41b4-000af7992758/practical_mestorf-4473f943087b.vmx)
2017-12-14T03:01:51.168Z verbose vpxa[7F12B70] [Originator@6876 sub=Default opID=WFU-588c74c] [VpxaMoVm] static memory overhead for /vmfs/volumes/vsan:52b432a675d2bf59-7168e072a499aad9/98e9315a-e2cf-0d9c-41b4-000af7992758/practical_mestorf-4473f943087b.vmx changed from <unset> to 0
```
[snip]
```
2017-12-14T03:01:51.673Z verbose vpxa[7EF1B70] [Originator@6876 sub=vpxaMoVMotion opID=lro-1-77a4044d-8633-01-01-01-27] [InitiateDestination] (7716914965410412045) Finish with dstId 44173783
2017-12-14T03:01:51.673Z info vpxa[7EF1B70] [Originator@6876 sub=vpxLro opID=lro-1-77a4044d-8633-01-01-01-27] [VpxLRO] -- FINISH lro-1316683
```

**Comment by hickeng** [comment]
@jzt Do you have the full portlayer logs for this issue still? I'm curious whether the VM Remove event can be distinguished from an actual deletion.

There are two VC behaviours here that we need to deal with. It's unclear whether they are bugs or not:
1. WaitForSession - assuming that we can confirm the cVM started cleanly (need tether.debug) this needs to be handled. The implication here is that a migration event or a remove event can/will prevent an in-flight property collector request from receiving updates. This can be handled in two ways, and we should likely do both.
  a. Re-request final state on property collector timeout - this would get us the updated state which wasn't reported via the collector.
  b. detect the circumstance which invalidates the collector and re-issue the collector request for the time remaining in the operation.
2. Incorrect reporting of VM Removed via an event - ideally we can filter the event in some manner (maybe we detect the Added event for an already known VM and expect a corresponding false Remove), however if that's not possible then we will have to treat the events as hints rather than definitive and confirm the VM has been removed from vCenter inventory

**Comment by jzt** [comment]
@lcastellano logs are attached
[vic-longevity-test-output-2017-12-13T13_23-06_00.tar.gz](https://github.com/vmware/vic/files/1612778/vic-longevity-test-output-2017-12-13T13_23-06_00.tar.gz)

**Comment by hickeng** [comment]
@jzt Did you determine if a regular vMotion performs the Remove/VmAdded tuple?

**Comment by jzt** [comment]
@hickeng in all three test cases, I observed the remove/vmAdded events. Here is an example:

dest VM vpxa.log:
```
2018-01-08T22:23:17.251Z verbose vpxa[57C7B70] [Originator@6876 sub=halservices opID=WFU-26ef851d] [VpxaHalServices] VmAdded Event for vm(0) ds:///vmfs/volumes/vsan:52d68b37d1c09c81-e3c6ac5092f4cf5717d4535a-0074-fd4b-6342-02000b66cd44/infallible_elion-e33722c0b9e6.vmx
```
source VM vpxa.log:
```
2018-01-08T22:23:33.305Z verbose vpxa[643FB70] [Originator@6876 sub=halservices opID=WFU-1240025] [VpxaHalServices] VmRemoved Event for vm(3) 2
```
source VM hostd.log:
```
2018-01-08T22:23:33.300Z info hostd[F0C2B70] [Originator@6876 sub=Vimsvc.ha-eventmgr] Event 542 : Removed infallible_elion-e33722c0b9e6 on sc-rdops-vm06-dhcp-190-105.eng.vmware.com from ha-datacenter
```

I did notice the ignored VmRemove event on the source host as well:
```
2018-01-08T22:23:33.307Z verbose vpxa[643FB70] [Originator@6876 sub=hostdevent opID=WFU-1240025] Event vim.event.VmRemovedEvent ignored: invalid vm argument (1)
```

Perhaps it detected it as a duplicate remove event? The `opID` and thread ID are the same. Does it make sense that we wouldn't see the remove event from `ha-eventmgr` and we miss the one from `hostdevent` since it gets ignored?

**Comment by lcastellano** [comment]
I have looked at the logs of the Longevity failure again and I don't think the problem is related to the events received after starting the VM. Rather it looks like the property collector (used in the WaitForSession call) is not returning timely after the VM has successfully started.  Here is the new issue: https://github.com/vmware/vic/issues/7098

**Comment by hickeng** [comment]
Need to determine if this still occurs. There is a CI test for `--rm` but is for a singular container and does not test any degree of concurrency.

**Comment by cgtexmex** [comment]
FWIW I just complete the following with no failures:

Run Command
`docker run -d --name $1 --rm --cpuset-cpus 1 --memory 1GB busybox /bin/sh -c 'a=0; while [ $a -lt 30 ]; do echo "line $a"; a=`expr $a + 1`; sleep 2; done;'`

Test 1 - 15 simultaneous deploys
Test 2 - 20 simultaneous deploys
Test 3 - 30 simultaneous deploys

Previously (1.3?) I was able to orphan containers with as few as 12 simultaneous 

Environment
3 Node Nimbus Cluster
6.5u2
NFS Storage

Code
Robo Feature branch -- including the **DRS PowerOn**

**Comment by cgtexmex** [comment]
Another data point:  ran the same test w/1.3.1 and it orphaned a container w/15 simultaneous deploys

**Comment by mhagen-vmware** [comment]
@hickeng the claim is that this was fixed between 1.3.1 and 1.4.0 (don't know what commit actually did it).  We have a current test that is passing (consistently).  Unless you really have a reason to have this open, I don't see any reason it should be.

**Comment by mhagen-vmware** [comment]
I have seen this fail twice now in the last couple of weeks, here is one of the builds:

https://ci-vic.vmware.com/vmware/vic/19274/7

The test was actually not working correctly as well, which may or may not have added to how rare it has become to see, but the failures I have seen are still valid regardless of the the test implementation. I have fixed the test here:

https://github.com/vmware/vic/pull/8017

**Comment by mhagen-vmware** [comment]
https://ci-vic.vmware.com/vmware/vic/19303

**Comment by hickeng** [comment]
This was observed again with @malikkal in May - also noted in https://github.com/vmware/vic/issues/7322#issuecomment-385922521 as it's unclear which path led us to the orphaned case.

**Comment by zjs** [comment]
I think this is worth a release note, as it was a customer-reported issue.

**Comment by stuclem** [comment]
Added the following as a resolved issue in https://github.com/vmware/vic/releases/tag/v1.4.3

- <b>Concurrent container auto remove orphans containers on vSphere. [#6342](https://github.com/vmware/vic/issues/6342)</b><br>Running concurrent containers with `docker run -d --rm` does not remove all container VMs after the run. Additionally, containers can be orphaned on vSphere, meaning that they are removed from the Focker persona, but the container VM remains in the vCenter Server inventory.

@zjs does this cover it? Thanks!
